Geometry/EcalAlgo/src/EcalEndcapGeometry.cc

```diff
@@ -516,10 +516,20 @@ EcalEndcapGeometry::avgAbsZFrontFaceCenter() const
    return m_avgZ;
 }
 
-const CaloCellGeometry* 
-EcalEndcapGeometry::cellGeomPtr( uint32_t index ) const
-{
-   const CaloCellGeometry* cell ( &m_cellVec[ index ] ) ;
-   return ( m_cellVec.size() < index ||
-	    nullptr == cell->param() ? nullptr : cell ) ;
+const CaloCellGeometry*
+EcalEndcapGeometry::cellGeomPtr(uint32_t index) const {
+  const CaloCellGeometry* cell(&m_cellVec[index]);
+  return ( m_cellVec.size() < index ||
+	   nullptr == cell->param() ? nullptr : cell ) ;
+}
+
+std::shared_ptr<CaloCellGeometry>  
+EcalEndcapGeometry::cellGeomPtr( uint32_t index ) {
+  if (m_cellVec.size() < index) return nullptr;
+  const auto do_not_delete = [](const void*){};
```

Comment:
If you make this `static const auto ...` it will be initialised only once per job, rather than once per call.

Reply:
Done